### DISCUSSION TITLE
Build Wheel for Mac OS M1 Silicon

### DISCUSSION THREAD

**Discussion by Navbryce** [sqlalchemy/sqlalchemy#8521] [Ideas]
Currently, there is no [wheel](https://pypi.org/project/SQLAlchemy/1.4.41/#files) for Mac OS M1 Silicon. Is it possible it could be added to the [build wheels action](https://github.com/sqlalchemy/sqlalchemy/blob/main/.github/workflows/create-wheels.yaml) ([relevant docs about cibuildwheel these docs](https://cibuildwheel.readthedocs.io/en/stable/faq/))?

I think you just need to add to [the env here](https://github.com/sqlalchemy/sqlalchemy/blob/main/.github/workflows/create-wheels.yaml#L63):
```CIBW_ARCHS_MACOS: x86_64 arm64```

I would make this change myself, but I lack the context on the build system and the library itself, 😢 sorry

**Comment by CaselIT** [comment]
Hi,

Version 2 adopts the cibuildwheel system supported by the pypi community so it will have arm mac builds.
The file you linked to is the new one that will be used once b2 starts shipping, and it does already build for arm. A test of it was done here https://test.pypi.org/project/SQLAlchemy/2.0.0b1.dev0/#files (note that it's a random commit of the v2 branch, not a release in any way, since v2 is still under active development)

Since version 1.4 still supports python 2 and cibuildwheel no longer supports it we opted for keeping the current build system that sadly does not have arm builds for mac.

**Reply by Navbryce** [reply]
Thanks for the update! Makes sense! 🤞 on the V2 release

**Reply by CaselIT** [reply]
the current eta is to have betas out in the fall and hopefully a first version by the end of the year, start of the new year
